src/lib/validering.ts

```diff
@@ -13,21 +13,22 @@ export const validateStringAsPositiveNumber = yup
     .moreThan(0, 'Feltet må være et positivt tall høyere enn 0')
     .typeError('Feltet må være et tall') as unknown as yup.Schema<string>;
 
-export const validateStringAsNonNegativeNumber = yup
-    .number()
-    .required('Feltet må fylles ut')
-    .min(0, 'Feltet må være større eller lik 0')
-    .typeError('Feltet må være et tall') as unknown as yup.Schema<string>;
+export function validateStringAsNonNegativeNumber(name = 'feltet') {
+    // Vi ønsker at tom streng skal regnes som at feltet ikke er fylt inn,
+    // men yup.number() vil behandle det som et ugyldig tall.
+    return yup.lazy((val) =>
+        val === ''
+            ? yup.string().required().label(name)
```

Comment:
Hva skjer dersom vi ikke tilfredstiller "required" delen her? Blir den validert av en annen funksjon før vi kommer hit? I så fall, trenger vi required her?

Reply:
Hvis vi ikke tilfredstiller `required` (`val` er `undefined`) så vil vi få feilmeldingen om at feltet er påkrevd. Var det det du tenkte på?

Reply:
Det jeg mente var at funksjonen ikke nevner noe om at den krever at feltet er påkrevd. Vi kunne kanskje hatt tilfeller der vi ville validere et felt som tall, men ikke som påkrevd. I tillegg har vi fjernet feilmeldingen for required, så jeg tenkte en annen funksjon sikret den valideringa? Men jeg ser den brukes som eneste funksjon, så da har du sikkert bare fjernet feilmeldingsteksten siden den var redundant med den som er default.

Reply:
Stemmer at required-meldinga var redundant med default ja. Funksjonen nevner at feltet er påkrevd i den grad at den gir `yup.Schema<string>` og ikke f.eks. `yup.Schema<string | null>` e.l. Hvis det blir et behov senere så kan vi heller ta tak i det da :)